### ISSUE TITLE
Poetry builds wheel package with wrong version

### ISSUE THREAD

**Issue by mshenhera** [sdispater/poetry#1092]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Debian GNU/Linux 9
- **Poetry version**: 0.12.15
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/mshenhera/ced9c3f98d494eb435552eec5b7b3d00

## Issue
When package version has local version identifiers (https://www.python.org/dev/peps/pep-0440/#local-version-identifiers) for example 0.1.2+77cc516 poetry creates wheel package with wrong version: replaces "+" by "_". Steps to reproduce the problem:
```sh
# Install poetry, jq, wheel-inspect
# Clone simple python project 
git clone https://github.com/pypa/sampleproject.git
cd sampleproject
# Create pyproject.toml  from github gist
curl -o pyproject.toml https://gist.githubusercontent.com/mshenhera/ced9c3f98d494eb435552eec5b7b3d00/raw/c33b8107773ca549fab72255527440f4492617e8/gistfile1.txt
# Build wheel package
~/.poetry/bin/poetry build -f wheel
# Inspect the package
wheel2json dist/sampleproject-0.1.2_77cc516-py3-none-any.whl  | jq .version
"0.1.2_77cc516"

# Build wheel package with setutools to show correct behavior:
# Change version in setup.py to something like "0.1.3+77cc516"
python setup.py bdist_wheel
wheel2json dist/sampleproject-0.1.3+77cc516-py2.py3-none-any.whl | jq .version
"0.1.3+77cc516"
```

As result of wrong version generation it is not possible to install package after uploading it to pypi repo. Uploads works fine. However Install fails with error:
```
ERROR: Could not find a version that satisfies the requirement sampleproject==0.1.2_77cc516
```

poetry generates poetry.lock with content like this one:
```
[[package]]
category = "main"
description = "A sample Python project"
name = "sampleproject"
optional = false
python-versions = "*"
version = "0.1.20_dc43e19"
```

After changed version in poetry.lock manually from 0.1.20_dc43e19 to 0.1.20+dc43e19 I was able to install the package.

**Comment by chrahunt** [comment]
Can you please provide the `pyproject.toml` for the project that is trying to use `sampleproject`?

In [PEP 427](https://www.python.org/dev/peps/pep-0427/) it states (regarding wheel filenames):

> Each component of the filename is escaped by replacing runs of non-alphanumeric characters with an underscore `_`:
>
>     re.sub("[^\w\d.]+", "_", distribution, re.UNICODE)

So Poetry is doing the right thing per the PEP. But in the `pyproject.toml` you must still specify the version like `<public version>+<local version>`.

**Comment by mshenhera** [comment]
pyproject.toml: https://gist.github.com/mshenhera/ced9c3f98d494eb435552eec5b7b3d00.

PEP 427 specifies filename format that does nothing with version, I guess. And there is no problem with filename as I can see. Poetry can install it without any issue as soon as version is manually corrected in poetry.lock.

**Comment by kbakk** [comment]
This issue is related to #756 (that goes on the file name causing issues). 

@chrahunt I do find PEP 427 interesting though, hadn't seen that when implementing PR (#950) to fix #756. 

As can be seen from @mshenhera's post, the filename setuptools uses contains `+`.

> ```
> ~/.poetry/bin/poetry build -f wheel
> ...
> wheel2json dist/sampleproject-0.1.2_77cc516-py3-none-any.whl
> ...
> python setup.py bdist_wheel
> wheel2json dist/sampleproject-0.1.3+77cc516-py2.py3-none-any.whl
> ...
> ```

Given that the current behavior breaks multiple implementations (as mentioned in #756), I think it makes sense to _do as the Romans do_ - i.e. produce same result as setuptools.

**Comment by kbakk** [comment]
Ok, looking at how wheel-inspect extracts version from filename, not the wheel itself ([wheel_inspect/filename.py#L13](https://github.com/jwodder/wheel-inspect/blob/v1.3.0/src/wheel_inspect/filename.py#L13)), I believe this issue is a duplicate of #756. 

The discussion here is interesting though, and a decision on how to go about filenames should be taken.
